src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.cs

```diff
@@ -0,0 +1,280 @@
+        }
+
+        private string GetIsThemeDefaultedKeyName(Guid themeId)
+        {
+            return $"IsThemeDefaulted.{themeId}";
+        }
+
+        private void SetColorScheme(string colorSchemeName)
+        {
+            // Currently, this is the only subsitition token supported here.
+            const string RootKeyToken = "$RootKey$";
+
+            using var registryRoot = VSRegistry.RegistryRoot(_serviceProvider, __VsLocalRegistryType.RegType_Configuration, writable: true);
+
+            // The below code is not a general purpose PkgDef merge utility, but one that only
+            // supports the "color theme" format.
+            using (var colorSchemeStream = GetColorSchemePackageDefStream(colorSchemeName))
+            using (var reader = new PkgDefFileReader(new StreamReader(colorSchemeStream)))
+            {
+                PkgDefItem? item;
+                while ((item = reader.Read()) != null)
+                {
+                    if ((item.ValueType == PkgDefItem.PkgDefValueType.Binary ||
+                            item.ValueType == PkgDefItem.PkgDefValueType.DWord) &&
+                        item.SectionName.StartsWith(RootKeyToken) && item.SectionName[RootKeyToken.Length] == '\\')
+                    {
+                        var itemRoot = item.SectionName.Substring(RootKeyToken.Length + 1);
+                        using var itemKey = registryRoot.CreateSubKey(itemRoot);
+                        itemKey.SetValue(item.ValueName, item.ValueData);
+                    }
+                }
+            }
+
+            // Set the current Color Scheme in the registry
+            using (var itemKey = registryRoot.CreateSubKey(RoslynTextEditorRegistryKey))
+            {
+                itemKey.SetValue(AppliedColorSchemeKeyName, colorSchemeName);
+            }
+        }
+
+        private Stream GetColorSchemePackageDefStream(string colorSchemeName)
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            return assembly.GetManifestResourceStream($"Microsoft.VisualStudio.LanguageServices.{colorSchemeName}.pkgdef");
+        }
+
+        private string GetCurrentColorScheme()
+        {
+            using var registryRoot = VSRegistry.RegistryRoot(_serviceProvider, __VsLocalRegistryType.RegType_Configuration, writable: true);
+            using var textEditorKey = registryRoot.CreateSubKey(RoslynTextEditorRegistryKey);
+            return (string)textEditorKey.GetValue(AppliedColorSchemeKeyName, defaultValue: string.Empty);
+        }
+
+        public bool IsKnownTheme()
+        {
+            return IsKnownTheme(GetThemeId());
+        }
+
+        public bool IsThemeCustomized()
+        {
+            return !_colorApplier.AreForegroundColorsDefaultable(GetThemeId());
+        }
+
+        public static bool IsKnownTheme(Guid currentTheme)
+        {
+            return currentTheme == KnownColorThemes.Light ||
```

Comment:
switch might be easier here

Reply:
Can only use switch expression with constants. KnownColorThemes values are all static readonly Guids.